pkg/services/dashboards/service/dashboard_service.go

```diff
@@ -415,6 +417,22 @@ func (dr *DashboardServiceImpl) SaveDashboard(ctx context.Context, dto *dashboar
 
 	return dash, nil
 }
+func (dr *DashboardServiceImpl) GetSoftDeletedDashboard(ctx context.Context, orgID int64, uid string) (*dashboards.Dashboard, error) {
+	return dr.dashboardStore.GetSoftDeletedDashboard(ctx, orgID, uid)
+}
```

Comment:
Both callers (api and service) are behind the feature flag. I could add the feature flag here as well and return an error if someone tries to call it but I would prefer it instead of creating a different service only for checking the flag.